core/src/test/java/io/questdb/test/std/datetime/millitime/DateFormatCompilerTest.java

```diff
@@ -155,6 +184,21 @@ public void testFormatEra() throws Exception {
         assertFormat("BC", "G", "-1024-04-09T00:00:00.000Z");
     }
 
+    @Test
+    public void testFormatFirstHourIn12HourClock() throws Exception {
+        assertFormat("04/09/2017 00:01:00 am", "MM/dd/yyyy KK:mm:ss a", "2017-04-09T00:01:00.000Z");
+        assertFormat("04/09/2017 00:59:59 am", "MM/dd/yyyy KK:mm:ss a", "2017-04-09T00:59:59.000Z");
```

Comment:
00 am looks invalid to me

Reply:
@ideoma: thanks for reviewing it! 

The pattern in this test is `KK:mm:ss a`
`KK` stands for `Hour in am/pm (0-11)`.
`a` stands for `am/pm`
=> It is correct, even uncommon. 

Normally the 12-hour clock uses the 1-12 convention. So that would change the placeholders from `KK:mm:ss a` to `hh:mm:ss a`. I can add this test too. wdyt?